src/bin/hexyl.rs

```diff
@@ -267,6 +305,50 @@ fn run() -> Result<(), AnyhowError> {
         .transpose()?
         .unwrap_or(0);
 
+    let columns = if let Some(columns) = matches
+        .value_of("columns")
+        .map(|s| {
+            s.parse::<NonZeroU8>().map(u8::from).context(anyhow!(
+                "failed to parse `--columns` arg {:?} as unsigned nonzero integer",
+                s
+            ))
+        })
+        .transpose()?
+    {
+        columns
+    } else {
+        let terminal_width = if let Some(terminal_width) = matches
+            .value_of("terminal_width")
+            .map(|s| {
+                s.parse::<NonZeroU16>().map(u16::from).context(anyhow!(
+                    "failed to parse `--columns` arg {:?} as unsigned nonzero integer",
+                    s
+                ))
+            })
+            .transpose()?
+        {
+            Some(terminal_width)
+        } else if matches.is_present("auto_width") {
+            Some(terminal_size().expect("not a tty").0 .0)
```

Comment:
I think we should properly handle this case (not a tty), by which I mean: return a proper error here and show a nice error message for the user. This can easily be triggered if the output of hexyl is being redirected.